snapshots.go

```diff
@@ -0,0 +1,136 @@
+package godo
+
+import "fmt"
+
+const snapshotBasePath = "v2/snapshots"
+
+// SnapshotsService is an interface for interfacing with the snapshots
+// endpoints of the DigitalOcean API
+// See: https://developers.digitalocean.com/documentation/v2#snapshots
+type SnapshotsService interface {
+	List(*ListOptions) ([]Snapshot, *Response, error)
+	ListVolume(*ListOptions) ([]Snapshot, *Response, error)
+	ListDroplet(*ListOptions) ([]Snapshot, *Response, error)
+	Get(string) (*Snapshot, *Response, error)
+	Delete(string) (*Response, error)
+}
+
+// SnapshotsServiceOp handles communication with the snapshot related methods of the
+// DigitalOcean API.
+type SnapshotsServiceOp struct {
+	client *Client
+}
+
+var _ SnapshotsService = &SnapshotsServiceOp{}
+
+// Snapshot represents a DigitalOcean Snapshot
+type Snapshot struct {
+	ID            string   `json:"id,omitempty"`
+	Name          string   `json:"name,omitempty"`
+	ResourceID    string   `json:"type,omitempty"`
+	ResourceType  string   `json:"type,omitempty"`
+	Regions       []string `json:"regions,omitempty"`
+	MinDiskSize   int      `json:"min_disk_size,omitempty"`
+	SizeGigabytes int      `json:"size_gigabytes,omitempty"`
+	Created       string   `json:"created_at,omitempty"`
+}
+
+type snapshotRoot struct {
+	Snapshot Snapshot
+}
+
+type snapshotsRoot struct {
+	Snapshots []Snapshot
+	Links     *Links `json:"links"`
+}
+
+type listSnapshotOptions struct {
+	ResourceType string `url:"resource_type,omitempty"`
+}
+
+func (s Snapshot) String() string {
+	return Stringify(s)
+}
+
+// List lists all the snapshots available.
+func (s *SnapshotsServiceOp) List(opt *ListOptions) ([]Snapshot, *Response, error) {
+	return s.list(opt, nil)
+}
+
+// ListDroplet lists all the Droplet snapshots.
+func (s *SnapshotsServiceOp) ListDroplet(opt *ListOptions) ([]Snapshot, *Response, error) {
+	listOpt := listSnapshotOptions{ResourceType: "droplet"}
+	return s.list(opt, &listOpt)
+}
+
+// ListVolume lists all the volume snapshots.
+func (s *SnapshotsServiceOp) ListVolume(opt *ListOptions) ([]Snapshot, *Response, error) {
+	listOpt := listSnapshotOptions{ResourceType: "volume"}
```

Comment:
Oh, hm, https://github.com/digitalocean/godo/blob/master/storage.go#L153-L172 maybe should be removed as dupe.